packages/abi-coder/src/encoding/coders/v0/ByteCoder.ts

```diff
@@ -42,8 +38,9 @@ export class ByteCoder extends Coder<number[], Uint8Array> {
     return concatWithDynamicData(parts);
   }
 
-  #getPaddedData(value: number[]): Uint8Array {
-    const data: Uint8Array[] = [Uint8Array.from(value)];
+  #getPaddedData(value: number[] | Uint8Array): Uint8Array {
+    const bytes = value instanceof Uint8Array ? value : new Uint8Array(value);
+    const data: Uint8Array[] = [Uint8Array.from(bytes)];
```

Comment:
`bytes` is already a `Uint8Array`, is it not?

```suggestion
    const bytes = value instanceof Uint8Array ? value : new Uint8Array(value);
    const data: Uint8Array[] = [bytes];
```

Reply:
Good spot.

Reply:
https://github.com/FuelLabs/fuels-ts/pull/1832/commits/204f4810acd7045110e2d73832c6091f2a93259f